code/modules/mob/living/simple_animal/hostile/megafauna/colossus.dm

```diff
@@ -734,7 +735,7 @@ Difficulty: Very Hard
 
 /obj/structure/closet/stasis/process()
 	if(holder_animal)
-		if(holder_animal.stat == DEAD && !QDELETED(holder_animal))
```

Comment:
If the mob it's in gets Qdeleted, it qdeletes its contents. Closets that get qdeleted drop their contents. If it fails to fire because a mob is qdeleted, it fails to assign the ghost back to the player's corpse, IE: You get stepped on as a cockroach and go splat, your body falls out but your ghost isn't able to get back in your body.

Basically, checking if a mob is qdeleted is entirely irrelevant because it can't be qdeleted before running this code.